app/services/submission_carousel.rb

```diff
@@ -0,0 +1,27 @@
+class SubmissionCarousel
+  FILTERS = [:valid, :rejected, :to_rate, :accepted, :waitlist, :with_confirmed_invitation]
+
+  def self.build(filter)
+    raise ArgumentError unless FILTERS.include?(filter)
```

Comment:
👍  You could probably add some message, to make it easy for debugging later, but good that this check is here.